docs/proposals/local-cluster.md

```diff
@@ -0,0 +1,486 @@
+plugin to use. When `bridge` plugin is used, `local-cluster` joins the
+bridges from the nodes together with another bridge. Other supported
+CNI implementations are `weave`, `flannel` and `calico`.
+
+`NO_DEDICATED` (`--no-dedicated`, not set by default) removes
+`dedicated` taint from master node even if `NUM_NODES` is greater than
+zero.
+
+`DIND_SUBNET` (`--dind-subnet`, default: `10.192.0.0/16`) specifies a
+CIDR notation IP range for use by node containers.
+
+`APISERVER_PORT` (`--apiserver-port`, default: `8080`) specifies
+apiserver port to use. If you're working with remote Docker engine,
+you need to forward this port to `localhost` over ssh.
+
+`BUILD_KUBEADM` (`--build-kubeadm`, not set by default) instructs
+`local-cluster` to build `kubeadm` locally from Kubernetes sources.
+
+`BUILD_HYPERKUBE` (`--build-hyperkube`, not set by default) instructs
+`local-cluster` to build `hyperkube` binary locally from Kubernetes
+sources.
+
+`BUILD_KUBECTL` (`--build-kubectl`, not set by default) instructs
+`local-cluster` to build `kubectl` for user's OS and place it in
+`~/.local-cluster`.
+
+`KUBEADM_URL` (`--kubeadm-url`, not set by default) specifies
+an URL to download `kubeadm` binary from.
+
+`KUBEADM_SHA1` (`--kubeadm-sha1`, not set by default) specifies
+sha1sum of the kubeadm binary.
+
+`HYPERKUBE_URL` (`--hyperkube-url`, not set by default) specifies
+the URL to download `hyperkube` binary from.
+
+`HYPERKUBE_SHA1` (`--hyperkube-sha1`, not set by default) specifies
+sha1sum of `hyperkube` binary.
+
+`KUBECTL_URL` (`--kubectl-url`, not set by default) specifies
+the URL to download `kubectl` binary from. The binary will be used
+on user's machine and must match its OS and architecture.
+
+`KUBECTL_SHA1` (`--kubectl-sha1`, not set by default) specifies
+sha1sum of `kubectl` binary.
+
+`LOCAL_GO` (`--local-go`, not set by default) instructs
+`local-cluster` to use local Go installation to build
+`kubeadm`/`hyperkube`/`kubectl`. By default `local-cluster` uses k8s
+build container for this task. _NOTE:_ implementating this option may
+make the startup script more complex.
+
+`NO_DASHBOARD` (`--disable-dashboard`, not set by default) instructs
+`local-cluster` to skip installation of Kubernetes dashboard.
+
+`DASHBOARD_URL` (`--dashboard-url`, defaults to the URL of recent
+`yaml` file for Kubernetes dashboard) specifies the `yaml` file to
+use to install Kubernetes dashboard.
+
+`VERBOSE` (`--v`, defaults to `1`) specifies verbosity level for
+`local-cluster`. Set to at least `2` to see `kubeadm`
+output.
+
+In addition to this, `gce.sh` script bundled with `local-cluster` can
+be used to set up a cluster on a GCE instance that's prepared using
+`docker-machine`.
+
+## Implementation notes
+
+### Base image
+
+`local-cluster` by default uses a prebuilt image to start the cluster.
+There's an image per each supported Kubernetes version and supported
+architecture (`amd64`, `arm`, `arm64`). There's also an image for each
```

Comment:
Ok, although it'll probably require some effort on my side to set up VMs for these.